spec.md

```diff
@@ -100,6 +110,18 @@ interface ExpressionStatement <: Statement {
 
 An expression statement, i.e., a statement consisting of a single expression.
 
+## Directive
+
+```js
+interface Directive <: ExpressionStatement {
+    directive: string;
+}
+```
+
+A directive from the directive prologue of a script or function.
+The `directive` property is the raw string source of the directive.
+This source includes single or double wrapping quotes and may not contain an EscapeSequence or LineContinuation..
```

Comment:
> <code>raw</code> in fact is not needed when you have <code>directive</code> boolean flag which would simply state that 1) this <code>ExpressionStatement</code> contains string <code>Literal</code> inside of it and 2) <code>Literal.value</code> contains valid (unescaped) directive.

That's not backwards compatible because `Literal.value` _is_ escaped.

> Why do you think it wouldn't work in this case? They wouldn't be identical since your second example will not have <code>directive</code> flag (implicit <code>false</code>).

No, `"use\x20strict"` can be part of a Directive Prologue per [§14.1.1](http://www.ecma-international.org/ecma-262/6.0/#sec-directive-prologues-and-the-use-strict-directive) (and therefore its `ExpressionStatement` node needs a non-null `directive` property), it's just not a Use Strict Directive.

To provide a concrete example, the following programs have three identical directives (and in particular are both strict):
```js
"use strict"; "use\x20strict"; "yustrikt";
```
```js
"yustrikt"; "use\x20strict"; "use strict";
```

but these have no directives:
```js
; "use strict";
```
```js
("use strict")
```

ESTree needs to correctly capture the distinctions, which is not possible by adding only a boolean `directive` property.